src/core.c/Mu.rakumod

```diff
@@ -788,17 +788,23 @@ my class Mu { # declared in BOOTSTRAP
 
     proto method Numeric(|) {*}
     multi method Numeric(Mu:U \v:) {
-        warn "Use of uninitialized value of type {self.^name} in numeric context";
+        my $name = (defined($*VAR_NAME) ?? $*VAR_NAME !! try v.VAR.name) // '';
+        $name   ~= ' ' if $name ne '';
+        warn "Use of uninitialized value {$name}of type {self.^name} in numeric context";
```

Comment:
if you put the space *before*, you could do "value$name of" and lose the additional scope